src/node_sea.cc

```diff
@@ -235,10 +247,56 @@ std::optional<SeaConfig> ParseSingleExecutableConfig(
     result.flags |= SeaFlags::kDisableExperimentalSeaWarning;
   }
 
+  std::optional<bool> use_snapshot = parser.GetTopLevelBoolField("useSnapshot");
+  if (!use_snapshot.has_value()) {
+    FPrintF(
+        stderr, "\"useSnapshot\" field of %s is not a Boolean\n", config_path);
+    return std::nullopt;
+  }
+  if (use_snapshot.value()) {
+    result.flags |= SeaFlags::kUseSnapshot;
+  }
+
   return result;
 }
 
-ExitCode GenerateSingleExecutableBlob(const SeaConfig& config) {
+ExitCode GenerateSnapshotForSEA(const SeaConfig& config,
+                                const std::vector<std::string>& args,
+                                const std::vector<std::string>& exec_args,
+                                const std::string& main_script,
+                                std::vector<char>* snapshot_blob) {
+  SnapshotData snapshot;
+  std::vector<std::string> patched_args = {args[0], config.main_path};
```

Comment:
Should we also append the list of CLI args that comes after `node --experimental-sea-config sea-config.json`, i.e., all elements of `args` after index `0`, to this vector? I'm saying that because when we run `node --snapshot-blob snapshot.blob --build-snapshot snapshot.js a b c`, it persists `a`, `b` and `c` in the list of CLI args present during the snapshot building phase.
```suggestion
  std::vector<std::string> patched_args = args;
  patched_args.insert(patched_args.begin() + 1, config.main_path);
```

Reply:
> it persists a, b and c in the list of CLI args present during the snapshot building phase.

It probably does not matter because we will always refresh the arguments after the snapshot is deserialized. But I think it might be useful to pass the arguments for testing purposes (usually users should not configure any of them during snapshot building, we generally refuse to load the snapshot if the flags do not match, except for certain known safe flags).

Reply:
> It probably does not matter because we will always refresh the arguments after the snapshot is deserialized.

Okay, I had assumed that it was expected behavior because earlier I had come across a blog post (unofficial) that tried to store some of the CLI args - https://blog.logrocket.com/snapshot-flags-node-js-v18-8/

Reply:
Thinking about it again this seems to be a bit weird because the main script is configured in the config, while the command line arguments are passed from the command line. I think we should probably add a programmatic API or a JSON config field for the arguments instead, but that probably can also be useful for SEA without snapshot, so I'll leave this as a TODO.